File: LeetCode/744寻找比目标字母大的最小字母.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int left = 0, right = letters.length - 1;
        if (target >= letters[right]) {
            return letters[0];
        }
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (letters[mid] > target) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }
        return letters[left];
    }
}

File: LeetCode/744寻找比目标字母大的最小字母.go
func nextGreatestLetter(letters []byte, target byte) byte {
	if target >= letters[len(letters)-1] {
		return letters[0]
	}
	left := sort.Search(len(letters)-1, func(i int) bool {
		return letters[i] > target
	})
	return letters[left]
}